### DISCUSSION TITLE
Manual mounting of Graphql endpoints

### DISCUSSION THREAD

**Discussion by cssArchitect** [Papooch/nestjs-cls#132] [Q&A]
Hello, @Papooch. 

I have a question about setting up `CLS` in the middleware section. Specifically, I'm interested in setting up `CLS` for certain `GraphQL` operations (as an alternative to simple `HTTP` routes). Is there a way to do this? 

```
@Module({
    imports: [
        ClsModule.forRoot({
            global: true,
            middleware: { mount: false },
        }),
    ],
    // ...
})
export class AppModule implements NestModule {
    configure(consumer: MiddlewareConsumer) {
        consumer.apply(ClsMiddleware).forRoutes('custom/route') // here I would like to specify concrete Graphq operations, like `query Users {...}`;
    }
}
```
Thanks in advance!

**Comment by Papooch** [comment]
I'm afraid that's not possible, or at least not this simple. This way of binding middleware only works for HTTP routes.

If you want a more granular way, you'd have to create a custom middleware or interceptor or something that wraps the request, and based on some condition, wrap the rest of the execution flow with `cls.run` without using the built-in ClsMiddleware.

I would suggest that you look at the code of the `core` library under `cls-initializers` to get the idea.

But I'm afraid that this level of granularity of initializing the CLS context would only lead to confusion in the future, because you can't be sure when it's actually initialized.

**Comment by cssArchitect** [comment]
Yes, you are right, this isn't so trivial, especially in the case of middleware, since Nest only provides needed `ArgumentsHost` in guards, filters, and interceptors. Thank you, I will look for a custom implementation.
